pkg/domain/filters/containers.go

```diff
@@ -211,6 +211,31 @@ func GenerateContainerFilterFuncs(filter string, filterValues []string, r *libpo
 		}, nil
 	case "network":
 		return func(c *libpod.Container) bool {
+			networkMode, err := c.NetworkMode(nil)
+			if err != nil {
+				return false
+			}
+			// support docker like `--filter network=container:<IDorName>`
+			// check if networkMode is configured as `container:<ctr>`
+			// peform a match against filter `container:<IDorName>`
+			// networks is already going to be empty if `container:<ctr>` is configured as Mode
+			if strings.Contains(networkMode, "container:") {
```

Comment:
Why `strings.Contains()` instead of `strings.HasPrefix()`?

Reply:
As I look at this more - it really looks as though the check for `container:` needs to be integrated into the `filterValues` loop below, instead of done separately up here.

Reply:
This is resolved in latest commit.